client/app/bundles/course/level/pages/Level/index.jsx

```diff
@@ -0,0 +1,191 @@
+
+  handleLevelTextBlur() {
+    this.props.dispatch(sortLevels());
+  }
+
+  handleCreateLevel() {
+    return (e) => {
+      e.preventDefault();
+      if (!this.props.isLoading) {
+        this.props.dispatch(addLevel());
+      }
+    };
+  }
+
+  handleDeleteLevel(levelNumber) {
+    return (e) => {
+      e.preventDefault();
+      if (!this.props.isLoading) {
+        this.props.dispatch(deleteLevel(levelNumber));
+      }
+    };
+  }
+
+  handleSaveLevels() {
+    return (e) => {
+      e.preventDefault();
+      const successMessage = <FormattedMessage {...translations.saveSuccess} />;
+      if (!this.props.isLoading) {
+        this.props.dispatch(saveLevels(this.props.levels, successMessage));
+      }
+    };
+  }
+
+  renderBody() {
+    const rows = this.props.levels.slice(1).map((experiencePointsThreshold, index) => (
+      <LevelRow
+        key={index + 1}
+        levelNumber={index + 1}
+        experiencePointsThreshold={experiencePointsThreshold}
+        updateExpThreshold={this.handleUpdateExpThreshold}
+        sortLevels={this.handleLevelTextBlur}
+        deleteLevel={this.handleDeleteLevel}
+        isLoading={this.props.isLoading}
+      />
+    ));
+
+    return (
+      <div style={styles.body}>
+        <Table className="table levels-list">
+          <TableHeader adjustForCheckbox={false} displaySelectAll={false}>
+            <TableRow>
+              <TableHeaderColumn style={{ paddingLeft: '74px' }}><FormattedMessage {...translations.levelHeader} /></TableHeaderColumn>
+              <TableHeaderColumn style={{ paddingLeft: '48px' }}><FormattedMessage {...translations.thresholdHeader} /></TableHeaderColumn>
+            </TableRow>
+          </TableHeader>
+          <TableBody>
+            {rows}
+          </TableBody>
+          <TableFooter>
+            <TableRow>
+              <TableRowColumn colSpan="1" style={{ textAlign: 'center' }}>
+                <FlatButton
+                  icon={<i className="fa fa-plus" />}
+                  label={<FormattedMessage {...translations.addNewLevel} />}
+                  disabled={this.props.isLoading}
+                  onClick={this.handleCreateLevel()}
+                />
+              </TableRowColumn>
+            </TableRow>
+            <TableRow>
+              <TableRowColumn colSpan="1" style={{ textAlign: 'left' }}>
+                <RaisedButton
+                  style={styles.formButton}
+                  type="submit"
+                  label={<FormattedMessage {...translations.saveLevels} />}
+                  primary
+                  onClick={this.handleSaveLevels()}
+                />
+              </TableRowColumn>
+            </TableRow>
+          </TableFooter>
+        </Table>
+      </div>
+    );
+  }
+
+  render() {
+    return (
+      <div>
+        <TitleBar title={<FormattedMessage {...defaultComponentTitles.course_levels_component} />} />
+        { this.props.isLoading ? <LoadingIndicator /> : this.renderBody() }
```

Comment:
You actually don't have to check `isLoading` anywhere else in the code because if it is false none of the stuff in `renderBody` will be rendered.

Reply:
Removed `isLoading` check from the event handlers.